### PR TITLE
Add `/rebase` command

### LINKED ISSUE #10904: CI: GitHub comment-ops command for rebasing and squashing
We should offer a slash command in the GitHub comment ops (aka slash command dispatch) tools to rebase and to squash PRs to prepare them for merging.

### PR DESCRIPTION
**Pull request by aq17**
# Description

Fixes #10904

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Comment by aq17** [comment]
`/rebase` to rebase, `/autosquash` or `rebase-autosquash` to autosquash 
https://github.com/cirrus-actions/rebase

File: .github/workflows/rebase.yml
Comment (AaronFriel) [inline .github/workflows/rebase.yml:13]: I think this won't quite implement #10904, `autosquash` performs `git rebase --autosquash`, which doesn't squash _all_ commits.

File: .github/workflows/rebase.yml
Comment (AaronFriel) [inline .github/workflows/rebase.yml:10]: I think we'll want to restrict this here to collaborators (people who have push access) https://github.com/cirrus-actions/rebase#restricting-who-can-call-the-action

I believe the permission we want is "COLLABORATOR" not "MEMBER"

**Review by AaronFriel — CHANGES_REQUESTED** [review]
No issue with this PR, but we might want to expand on it or make an upstream PR to cirrus-actions to support a true "/squash" mode, that squashes all commits.

I think after restricting access, we can merge this and I can revisit #10904 this week

**Review by AaronFriel — APPROVED** [review]
LGTM!

**Comment by aq17** [comment]
bors merge

**Pull request merged**
